File: backend/src/main/java/nextpos/app/nextpos/service/interf/CompanyService.java
package nextpos.app.nextpos.service.interf;

import nextpos.app.nextpos.model.dto.request.CreateRequest.CreateCompanyRequest;
import nextpos.app.nextpos.model.dto.request.UpdateRequest.UpdateCompanyRequest;
import nextpos.app.nextpos.model.dto.response.CompanyResponse;

public interface CompanyService {

    /**
     * Create a new company.
     */
    CompanyResponse createCompany(CreateCompanyRequest request);

    /**
     * Update company fields (partial updates supported).
     */
    CompanyResponse updateCompany(Long companyId, UpdateCompanyRequest request);

    /**
     * Get company by id (with settings embedded).
     */
    CompanyResponse getCompany(Long companyId);

    /**
     * Soft-delete a company (mark as deleted + de-activate).
     */
    void deleteCompany(Long companyId);
}

File: frontend/src/services/companyService.ts
import api from "./api";
import type {
  CompanyListItem,
  CompanyDetail,
  CreateCompanyRequest,
  UpdateCompanyRequest,
} from "@/types/Company";

export const createCompany = (data: CreateCompanyRequest) =>
  api.post<CompanyDetail>("/companies", data);

export const updateCompany = (id: number, data: UpdateCompanyRequest) =>
  api.put<CompanyDetail>(`/companies/${id}`, data);

export const getCompanyById = (id: number) =>
  api.get<CompanyDetail>(`/companies/${id}`);

export const deleteCompany = (id: number) =>
  api.delete<void>(`/companies/${id}`);
